sdk/core/System.Net.ClientModel/src/ModelReaderWriter/ModelReaderWriter.cs

```diff
@@ -12,6 +13,21 @@ namespace System.Net.ClientModel.Core
     /// </summary>
     public static class ModelReaderWriter
     {
+        /// <summary>
+        /// Creates an instance of a model from an instance of an anonymous type.
+        /// </summary>
+        /// <typeparam name="T">The type of the model to create.</typeparam>
+        /// <param name="jsonSerializable">An instance of an anonymous type representing the model</param>
+        /// <returns>A <typeparamref name="T"/> instance.</returns>
+        public static T? Create<T>(object jsonSerializable) where T : IJsonModel<T>
+        {
+            BinaryData json = BinaryData.FromObjectAsJson(jsonSerializable);
+            var reader = new Utf8JsonReader(json);
+            IJsonModel<T> instance = (T)Activator.CreateInstance(typeof(T), true)!;
```

Comment:
We should work out the use case for a discriminated set.  If we ever want to do this with the base class we will need to call into the logic below that uses the proxy attribute.